### PR TITLE
Improve API

### PR DESCRIPTION
**Pull request by svenwltr**
@hendrik-rebuy @rebuy-de/it-platform

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,7 +94,19 @@
...
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>${powermock.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <version>${powermock.version}</version>
             <scope>test</scope>
```
Comment (florian-zeidler) [inline pom.xml:110]: Powermock to the rescue :hear_no_evil:

**Comment by florian-zeidler** [comment]
:+1:

**Comment by ghost** [comment]
:+1:

File: src/main/java/com/rebuy/consul/ConsulService.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    public void register()
+    {
+        client.agentServiceRegister(service);
+    }
+
+    public void shutdownHook()
+    {
+        Runtime.getRuntime().addShutdownHook(
+            new Thread(this::unregister));
```
Comment (ghost) [inline src/main/java/com/rebuy/consul/ConsulService.java:31]: java 8 :heart:

Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/ConsulService.java:31]: it is good to start a new thread on shutdown?

Comment (svenwltr) [inline src/main/java/com/rebuy/consul/ConsulService.java:31]: That's how the Java API works: https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-

Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/ConsulService.java:31]: :+1:

**Comment by bjoernhaeuser** [comment]
:+1:

**Pull request merged**